styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -121,17 +122,18 @@ public void restore(RunState runState, long count) {
   }
 
   @Override
-  public void receive(Event event) throws IsClosed {
+  public CompletionStage<Void> receive(Event event) throws IsClosed {
     ensureRunning();
 
     final InstanceState state = states.get(event.workflowInstance());
     if (state == null) {
       LOG.warn("Received event for unknown workflow instance: {}", event);
-      return;
+      return null;
```

Comment:
This will eventually propagate to `TriggerManager.tick()` where we do `processed.toCompletableFuture`, or? `CompletableFuture.completedFuture(null)` instead?